### ISSUE TITLE
shapes.Line x,y acts as an origin, and I don't see a way to edit the coords of a single point inside a MultiLine

### ISSUE THREAD

**Issue by jokoon** [pyglet/pyglet#1079]
I am making an untangle game, which looks like a bit like this

![image](https://github.com/pyglet/pyglet/assets/533869/d9775d74-bc47-4e55-bc9f-be50b885a204)

Basically, the game is about moving points that are connected to each other.

When I move the x,y of a Line with pyglet, it moves the entire line, while I want the x2 y2 to be an absolute position, not relative to x,y. I don't want x2 y2 to move if I change x y.

I have quickly tried making my own Line2 class, inheriting from ShapeBase, by copying code around, but it's missing functions that are in shapes.py.

It seems I can't really set or access a point of a MultiLine, although I could recreate the entire MultiLine object from scrach when I want edit a single point.

I am a bit confused if my use-case is a bit too particular, and in which direction I should go. SFML has a VertexArray that allows to edit any point directly. tkinter has lines, but tkinter is a bit awkward too.

I guess a solution would be either to add a way to edit the points of a multiline, and make Line points absolute and not relative.

I might be missing something.

**Comment by jokoon** [comment]
I was able to solve it with 

```
    shader_program = pyglet.shapes.get_default_shader()

    vertex_list = shader_program.vertex_list(
        # 4, GL_TRIANGLES, batch, group,
        count = len(flat) // 2,
        mode = pyglet.gl.GL_LINES,
        batch=batch_lines,
        group = None,
        position=('f', flat),
        colors=('Bn', [64, 64, 64, 255]*(len(flat)//2),)
        )```

I am not managing well to access the proper data with `vertex_list.position[]` for alignment but I will try later.

**Comment by benmoran56** [comment]
You can make it work with Line if you re-set the x2, and y2 properties afterwards. In this example, you can see that left-clicking always repositions the entire object, but right-clicking afterwards will allow you to reset the X2,Y2 positions:
```py
import pyglet

window = pyglet.window.Window()
batch = pyglet.graphics.Batch()

line = pyglet.shapes.Line(x=100, y=100, x2=400, y2=400, width=2, batch=batch)


@window.event
def on_mouse_press(x, y, button, mod):
    if button == pyglet.window.mouse.LEFT:
        line.position = x, y
    elif button == pyglet.window.mouse.RIGHT:
        line.x2 = x
        line.y2 = y


def update(dt):
    window.clear()
    batch.draw()


pyglet.clock.schedule_interval(update, 1/60)
pyglet.app.run()
```

**Comment by jokoon** [comment]
That code will "forget" the x2 y2 position when I click left.

A fix might be to recalculate new positions that match the previous x2 y2.

Apparently this fixes it:

```
@window.event
def on_mouse_press(x, y, button, mod):
    if button == pyglet.window.mouse.LEFT:
        x2 = line.x2
        y2 = line.y2
        line.position = x, y
        line.x2 = x2
        line.y2 = y2
    elif button == pyglet.window.mouse.RIGHT:
        line.x2 = x
        line.y2 = y
```

**Comment by benmoran56** [comment]
Yes that's right, my example was trying to show that it's a 2-step process. 

For the why, 
the reason that changing `x, y` moves the entire line is because it matches how all of the other shapes work. 
A common case is that someone defines a shape (like a many-point polygon), then they want to reposition it. They don't want to have to update _all_ of the points manually.
It admittedly does seem strange when working with a simple line, compared to other shapes (even Triangles) with multiple points, but that's why it behaves the way it does.

**Comment by caffeinepills** [comment]
Support for this has been added in pyglet 3.0 (currently development branch)
